### PR TITLE
Generalize "remove trivial selects"

### PR DESCRIPTION
**Pull request by rjmac**
Before it would consider removing a trivial select only if it were the _only_ input dataset to a query.  Now it will do for all trivial-select inputs even in the presence of joins.

### CODE REVIEW

**Review by saraid — APPROVED** [review]
You said something about SEARCH being a problem, but I'm not seeing it, so I'm approving.

File: soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/SoQLAnalyzerTest.scala [hunk 1]
```diff
@@ -758,6 +758,33 @@ class SoQLAnalyzerTest extends FunSuite with MustMatchers with TestHelper {
...
+
+  test("Hidden columns - union") {
+    val tf1 = tableFinder(
+      (0, "aaaa-aaaa") -> D("n1" -> TestNumber, "n2" -> TestNumber).withHiddenColumns("n1"),
+      (0, "bbbb-bbbb") -> D("n3" -> TestNumber),
+      (0, "cccc-cccc") -> Q(0, "aaaa-aaaa", "select * union select n3 from @bbbb-bbbb"),
+    )
+    val analysis1 = analyze(tf1, "cccc-cccc", "select :*, *")
+
+    println(analysis1.statement.debugDoc)
```
Comment (saraid) [inline soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/SoQLAnalyzerTest.scala:785]: Debug?

Comment (rjmac) [reply soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/SoQLAnalyzerTest.scala:785]: Yep, oops.  This entire class's change was not meant to be checked in.

**Review by saraid — APPROVED** [review]
I see no problems.

**Pull request merged**